lib/modules/manager/pipenv/artifacts.ts

```diff
@@ -120,6 +161,24 @@ export async function updateArtifacts({
         },
       ],
     };
+
+    const sourceUrl = await findPipfileSourceUrlWithCredentials(pipfileName);
+    if (sourceUrl) {
+      logger.debug('Pipfile contains credentials');
+      const hostRule = getMatchingHostRule(sourceUrl);
+      if (hostRule && execOptions.extraEnv) {
```

Comment:
move extraEnv to a separate const, so this check can be removed. extraEnv is always set.
```suggestion
      if (hostRule) {
```